File: backend/app/schemas/notification.py
from pydantic import BaseModel, Field
from uuid import UUID
from datetime import datetime
from typing import Optional
from app.models.enums import NotificationType

class NotificationResponse(BaseModel):
    id: UUID
    user_id: UUID
    tenant_id: UUID
    type: NotificationType
    message: str
    related_document_id: Optional[UUID] = None
    related_role_id: Optional[UUID] = None
    related_department_id: Optional[UUID] = None
    is_read: bool
    created_at: datetime

    model_config = {
        "from_attributes": True
    }

File: frontend/src/types/notification.ts
export type NotificationType =
  | 'role_assigned'
  | 'document_access_direct'
  | 'document_access_inherited_hierarchy'
  | 'document_access_inherited_department'
  | 'department_added'
  | 'evaluation_completed';

export interface Notification {
  id: string;
  user_id: string;
  tenant_id: string;
  type: NotificationType;
  message: string;
  related_document_id?: string;
  related_role_id?: string;
  related_department_id?: string;
  related_evaluation_id?: string;
  is_read: boolean;
  created_at: string;
}

